### ISSUE TITLE
submit Promise cannot have a promise inside of it?

### ISSUE THREAD

**Issue by bmueller-sykes** [erikras/redux-form#764]
Understanding that a big re-architect is coming, which therefore may make this issue moot, I figured I'd park it here just in case.

It appears that if the returned Promise of the `submit()` function contains a function that is itself a Promise, then I get an error. so, if I have something like:

```
submit = (values, dispatch) => {
    return new Promise ( (resolve, reject) => {
        thisFunctionReturnsAPromise().then(
            (response) => { resolve() },
            (error) => { reject() }
        )
    });
} 
```

...then processing is aborted somewhere (I know that the top-level `response` and `error` callback functions of the Promise do not appear to be invoked). When this happens, the error attached to the one field I have in the form says "ReferenceError" or "TypeError" (not sure what conditions bring up each of those cases), and so does the top-level form error.

If the `thisFunctionReturnsAPromise()` has no `then` statement, but rather just does this:

```
    thisFunctionReturnsAPromise();
    resolve();
```

...that works, but obviously, that's not what I want. So maybe I'm using redux-form wrong, or are doing something otherwise dumb, but in case this is a bug, I thought it was worth mentioning.

Thanks!

**Comment by ooflorent** [comment]
Thank you for reporting this issue!

I would be really grateful to you if you could submit a PR with a failing test thus we would be able to better identify the bug and fix it.

**Comment by bmueller-sykes** [comment]
apologies for not getting back to you. Once the new version of redux-forms is out, I'll try to replicate under your new architecture.

**Comment by ooflorent** [comment]
@bmueller-sykes It is out! It is available on npm under `6.0.0-alpha.13`

**Comment by bmueller-sykes** [comment]
sweet! Hopefully I can get to it in a week or so
